src/Caching/StackExchangeRedis/src/RedisCache.cs

```diff
@@ -257,6 +266,29 @@ private async Task ConnectAsync(CancellationToken token = default(CancellationTo
         }
     }
 
+    private void PrepareConnection()
+    {
+        ValidateServerFeatures();
+        TryRegisterProfiler();
+    }
+
+    private void ValidateServerFeatures()
+    {
+        if (_connection is null)
+        {
+            return;
+        }
+
+        foreach (var endPoint in _connection.GetEndPoints())
+        {
+            if (_connection.GetServer(endPoint).Version < ServerVersionWithExtendedSetCommand)
```

Comment:
Is there an async version of that?

Reply:
No, there are no async versions of that
[IConnectionMultiplexer.GetEndPoints() source](https://github.com/StackExchange/StackExchange.Redis/blob/f3052db96b071aa4bd88010ef2716d4d6f0329c7/src/StackExchange.Redis/Interfaces/IConnectionMultiplexer.cs#L118)
[IConnectionMultiplexer.GetServer() source](https://github.com/StackExchange/StackExchange.Redis/blob/f3052db96b071aa4bd88010ef2716d4d6f0329c7/src/StackExchange.Redis/Interfaces/IConnectionMultiplexer.cs#L190)

Reply:
After a quick glance, it looks like the version is [read from in-memory config](https://github.com/StackExchange/StackExchange.Redis/blob/f3052db96b071aa4bd88010ef2716d4d6f0329c7/src/StackExchange.Redis/ServerEndPoint.cs#L52) rather than queried from the server. Is that right, @NickCraver?